### ISSUE TITLE
Fix broken random.alphaNumeric() test

### ISSUE THREAD

**Issue by ST-DDT** [faker-js/faker#513]
https://github.com/faker-js/faker/blob/431c10887ae43c5ace1f653db91b5540c511db7e/test/random.spec.ts#L180-L187

https://github.com/faker-js/faker/blob/431c10887ae43c5ace1f653db91b5540c511db7e/src/random.ts#L383

It just has a very low chance of failing (~0.2%).

The test should be fixed like this:

````diff
- expect(alphaText).match(/[b-oq-z]/);
+ expect(alphaText).match(/^[0-9b-oq-z]{5}$/);
````

The same is true for the following test.
I also recommend to ban more chars, as the current test only has a chance of about ~25%(?) to actually the test the expected result.

_Originally posted by @ST-DDT in https://github.com/faker-js/faker/issues/508#issuecomment-1045997000_

**Comment by ST-DDT** [comment]
@Shinigami92 Do you want to fix it, or is it up for grabs?

**Comment by xDivisionByZerox** [comment]
I will do this. No problem.

**Comment by xDivisionByZerox** [comment]
Wouldn't it be easier to iterate over each banned character and expect the alpha text to not include it? Like this
```typescript
it('should be able to ban some characters', () => {
  const bannedChars = ['a', 'p'];
  const alphaText = faker.random.alphaNumeric(5, {
    bannedChars,
  });

  expect(alphaText).toHaveLength(5);
  for(const bannedChar of bannedChars) {
    expect(alphaText.includes(bannedChar)).toBe(false);
  }
});
```

This would allow to change the bannedChars Array without rewriting the RegEx.

**Comment by ST-DDT** [comment]
IMO we should ban all characters `a` - `p` to have a higher probability to "trigger" a banned character.

As for "foreach vs regex": I think it's fine either way.

@Shinigami92 Any preference?

**Comment by xDivisionByZerox** [comment]
We could also force test this by splitting this into 2 tests. One with all alphabetic characters banned and one with all numbers characters baned.

**Comment by ST-DDT** [comment]
I really like that idea.

`foreach` vs `regex`:
If it isn't too much work, start with both variants in after each other (in the same method). 
That makes comparing it easier.
Then we can decide on one of them easily afterwards.

**Comment by xDivisionByZerox** [comment]
Should the current test with some characters baned (instead of all alpha/number) still be reworked or replaced by the 2 tests I suggested?

Furthermore, what happens if a bannedChars Array with all alpha-numeric characters is provided? Should a test for that exist as well?

**Comment by xDivisionByZerox** [comment]
> foreach vs regex:
If it isn't too much work, start with both variants after each other (in the same method).
That makes comparing it easier.
Then we can decide on one of them easily afterwards.

Do you mean 2 tests:
- one for banned alpha, one for banned numbers where each test matches against regex AND foreach
OR
- one with regex matching against banned alpha AND numbers and one with foreach matching against banned alpha AND numbers

**Comment by ST-DDT** [comment]
The first one.

> one for banned alpha, one for banned numbers where each test matches against regex AND foreach

**Comment by ST-DDT** [comment]
> Furthermore, what happens if a bannedChars Array with all alpha-numeric characters is provided? Should a test for that exist as well?

Yes. Although I assume, that this will reveal another "bug", that we can only fix in v6.1. If that's the case disable it with `it.todo()`.
